### PR TITLE
Enhance SMT pattern installation case

### PR DESCRIPTION
**Pull request by Jiawei517**
Add mirror repo step after config SMT server.

See POO#28549

- Related ticket: https://progress.opensuse.org/issues/28549
- Verification run: http://10.67.17.147/tests/1752#step/smt/23

### CODE REVIEW

File: tests/x11/smt.pm [hunk 1]
```diff
@@ -57,6 +57,10 @@ sub run {
 
     assert_script_run 'smt-sync', 600;
     assert_script_run 'smt-repos';
+
+    # mirror and sync a base repo from SCC
+    assert_script_run 'smt-repos --enable-mirror SLES12-SP3-Installer-Updates sle-12-x86_64';
```
Comment (rwx788) [inline tests/x11/smt.pm]: I guess we should assume that test can be executed on any architecture. So I suggest that we use VERSION and ARCH here. We can skip sle 15 though, as RMT will replace SMT there.

**Comment by dehai** [comment]
could you package setting configure code into one sub and export it, because I will also use it in my disconnected SMT test case, please see my travis CI , it show much more duplicate with your this pm. 
https://travis-ci.org/dehai/os-autoinst-distri-opensuse/jobs/309433211
I prefer to package line 24 to 55.as its common code. thanks a lot.

File: tests/x11/smt.pm (hunk 1 above)
Comment (mitiao) [reply tests/x11/smt.pm]: smt pattern was only avaliable on x86_64 in SLE12 that means this test could be executed only on x86_64. I suggest we don't need to pay more attention on this since RMT test on SLE15 would be more important.

**Comment by mitiao** [comment]
@dehai  @Jiawei517 pair programming to extract common codes to functions :)

File: tests/x11/smt.pm (hunk 1 above)
Comment (rwx788) [reply tests/x11/smt.pm]: @mitio Absolutely. With SLE15 we will have separate test suite anyway. Thanks for clarification.

Comment (mitio) [reply tests/x11/smt.pm]: @rwx788 I think you meant @mitiao instead of @mitio :)

Comment (rwx788) [reply tests/x11/smt.pm]: @mitio True, sorry for that, tab suggested wrong person and I haven't recognized it =)

File: tests/x11/smt.pm [hunk 2]
```diff
@@ -54,10 +66,12 @@ sub run {
...
     send_key 'alt-n';
     wait_serial("yast2-smt-wizard-0", 400) || die 'smt wizard failed';
+}
 
-    assert_script_run 'smt-sync', 600;
-    assert_script_run 'smt-repos';
-    type_string "killall xterm\n";
+sub smt_mirror_repo {
+
+    assert_script_run 'smt-repos --enable-mirror SLES12-SP3-Installer-Updates sle-12-x86_64';
```
Comment (okurz) [inline tests/x11/smt.pm]: I'm concerned about that hardcoded string. Either use a test variable or use existing test variable settings to combine these strings or comment explicitly why it's ok to hardcode one version/distri/arch here

Comment (Jiawei517) [reply tests/x11/smt.pm]: I have add comment for it, but it was hidden after I extracted this mirror function.

Smt pattern is only avaliable on x86_64 in SLE12 that means this test could be executed only on x86_64. And this case is focus on testing the basic function of SMT pattern, so any released repo on SCC is ok for it. I suppose a tiny and stable repo would be better for this case.

RMT will replace SMT in SLES15, that means this case in only be executed with SLES12SP3 and SLES12SP4 in the future. I think it is ok to hardcdoed this string as SLES12-SP3-Installer-Updates repo here.

Comment (okurz) [reply tests/x11/smt.pm]: sure, just add a comment describing that and I am fine with the solution :)

**Comment by dehai** [comment]
@Jiawei517  I can't use your functions , could you move those two functions into lib directory.Thanks!
